### PR TITLE
Add more resources

### PR DESCRIPTION
**Pull request by calebbourg**
## Description
This PR adds APIs for the remaining resources for our initial PoC:
- `actions`
- `agreements`
- `overarching_goals`

It also updates a few tables and uses `user_id`s encoded in JWTs

### Changes
* Add remaining resources
* Update tables to add `status` and `status_changed_at`
* Use `user_id` from JWT

### Testing Strategy
tested using rapidoc

### CODE REVIEW

**Comment by calebbourg** [comment]
Here is the new PNG of the dbml diagram. I can never remember how to add it through the GitHub UI.

Untitled (1)

**Comment by jhodapp** [comment]
> Here is the new PNG of the dbml diagram. I can never remember how to add it through the GitHub UI.

I'll document that here: the PNG goes into src/entity/README.md. You can edit that file from GitHub itself within your PR here and drag and drop the updated PNG onto that file at the correct line within the file. GitHub uploads the updated diagram and creates a new markdown link to the file via a URI and UUID right where your cursor is.

File: web/src/controller/agreement_controller.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    get,
+    path = "/agreements/{id}",
+    params(
+        ApiVersion,
+        ("id" = String, Path, description = "Agreement id to retrieve")
+    ),
+    responses(
+        (status = 200, description = "Successfully retrieved a specific Agreement by its id", body = [entity::notes::Model]),
+        (status = 401, description = "Unauthorized"),
+        (status = 404, description = "Note not found"),
```
Comment (jhodapp) [inline web/src/controller/agreement_controller.rs:65]:
```suggestion
        (status = 404, description = "Agreement not found"),
```

File: web/src/controller/action_controller.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    get,
+    path = "/actions/{id}",
+    params(
+        ApiVersion,
+        ("id" = String, Path, description = "Action id to retrieve")
+    ),
+    responses(
+        (status = 200, description = "Successfully retrieved a specific Action by its id", body = [entity::notes::Model]),
+        (status = 401, description = "Unauthorized"),
+        (status = 404, description = "Note not found"),
```
Comment (jhodapp) [inline web/src/controller/action_controller.rs:62]:
```suggestion
        (status = 404, description = "Action not found"),
```

File: web/src/controller/overarching_goal_controller.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    get,
+    path = "/overarching_goals/{id}",
+    params(
+        ApiVersion,
+        ("id" = String, Path, description = "Overarching Goal id to retrieve")
+    ),
+    responses(
+        (status = 200, description = "Successfully retrieved a specific Overarching Goal by its id", body = [entity::notes::Model]),
+        (status = 401, description = "Unauthorized"),
+        (status = 404, description = "Note not found"),
```
Comment (jhodapp) [inline web/src/controller/overarching_goal_controller.rs:70]:
```suggestion
        (status = 404, description = "Overarching Goal not found"),
```

File: web/src/controller/overarching_goal_controller.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+use log::*;
+
+/// POST create a new Overarching Goal
+#[utoipa::path(
+    post,
+    path = "/overarching_goals",
+    params(ApiVersion),
+    request_body = entity::overarching_goals::Model,
+    responses(
+        (status = 201, description = "Successfully Created a New OverarchingGoal", body = [entity::overarching_goals::Model]),
```
Comment (jhodapp) [inline web/src/controller/overarching_goal_controller.rs:24]:
```suggestion
        (status = 201, description = "Successfully Created a New Overarching Goal", body = [entity::overarching_goals::Model]),
```

File: web/src/controller/overarching_goal_controller.rs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+)]
+pub async fn index(
+    CompareApiVersion(_v): CompareApiVersion,
+    AuthenticatedUser(_user): AuthenticatedUser,
+    // TODO: create a new Extractor to authorize the user to access
+    // the data requested
+    State(app_state): State<AppState>,
+    Query(params): Query<HashMap<String, String>>,
+) -> Result<impl IntoResponse, Error> {
+    debug!("GET all OverarchingGoals");
```
Comment (jhodapp) [inline web/src/controller/overarching_goal_controller.rs:152]:
```suggestion
    debug!("GET all Overarching Goals");
```

**Review by jhodapp — CHANGES_REQUESTED** [review]
Just some very minor string constant updates to fix and then I'll approve this PR. This is an exciting milestone of a PR!

**Review by jhodapp — APPROVED** [review]
Looks fabulous!

**Pull request merged**